### PR TITLE
Updates to ListAgents and ListTargets

### PR DESCRIPTION
**Pull request by johncmerfeld**
This change corresponds to the backend production release scheduled for 8/30

### CODE REVIEW

File: .changes/unreleased/FEATURES-20230815-152319.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+kind: FEATURES
+body: 'targets/disambiguated: Add a new service that consumes the disambiguated AllTargets
+  endpiont'
```
Comment (ymarcus93) [inline .changes/unreleased/FEATURES-20230815-152319.yaml:3]:
```suggestion
  endpoint'
```

File: bastionzero/service/targets_disambiguated/all_targets.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+func (t *Target) GetName() string                      { return t.Name }
+func (t *Target) GetStatus() targetstatus.TargetStatus { return t.Status }
+func (t *Target) GetEnvironmentID() string             { return t.EnvironmentID }
+func (t *Target) GetEnvironmentName() string           { return t.EnvironmentName }
+func (t *Target) GetAgent() *agents.AgentSummary       { return t.Agent }
+func (t *Target) GetAccessDetails() *AccessDetails     { return t.AccessDetails }
+
+type ListAllTargetsOptions struct {
+	// Each target's list of connections will include connections with only these states. Defaults to only Open.
+	ConnectionStates connectionstate.ConnectionState `url:"connectionStates,omitempty"`
```
Comment (ymarcus93) [inline bastionzero/service/targets_disambiguated/all_targets.go:90]:
```suggestion
	ConnectionStates []connectionstate.ConnectionState `url:"connectionStates,omitempty"`
```

I think this is a list on backend

File: bastionzero/service/targets_disambiguated/all_targets.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	GetLastAgentUpdate() *types.Timestamp
+	// GetAgentVersion returns the target's agent version.
+	GetAgentVersion() string
+	// GetRegion returns the target's region.
+	GetRegion() string
+	// GetAgentPublicKey returns the target's agent public key.
+	GetAgentPublicKey() string
+	// GetTargetType returns the target's type.
+	GetTargetType() targettype.TargetType
+}
```
Comment (ymarcus93) [inline bastionzero/service/targets_disambiguated/all_targets.go:60]: Doesn't look like this is being used by the new `_target.go` files in `targets_disambiguated`. Feel free to remove. In a future PR, we can bring this interface back and add common getters for the new disambiguated types.

File: .changes/unreleased/FEATURES-20230725-172355.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+kind: FEATURES
```
Comment (ymarcus93) [inline .changes/unreleased/FEATURES-20230725-172355.yaml:1]: This file can be deleted. Looks like a copy-paste from old PR

File: bastionzero/service/agents/agents.go [hunk 5]
```diff
@@ -57,6 +61,8 @@ type ListAgentsOptions struct {
 	EnvironmentName string `url:"environmentName,omitempty"`
 	// Only return agents whose name contains this substring. Note that terminated agents have no name.
 	Name string `url:"name,omitempty"`
+	// Only return agents with these UUIDs. Note that terminated agents have no ID.
+	AgentIds []string `url:"agentIds,omitempty"`
```
Comment (ymarcus93) [inline bastionzero/service/agents/agents.go:65]: Create another `changie` entry of `kind: Enhancements`:

`agents: Add ability to filter by agent ID when listing agents`

or something like that ^^ for entry

**Review by ymarcus93 — COMMENTED** [review]
Nice!

**Comment by johncmerfeld** [comment]
Thanks for the feedback. Believe I addressed everything in the latest commit

File: bastionzero/service/targets_disambiguated/all_targets.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+//
+// BastionZero API docs:
+//
+// - https://cloud.bastionzero.com/api
+
+type AllTargetsService client.Service
+
+type AccessDetails struct {
+	JIT                  bool            `json:"jit"`
+	AccessExpirationTime types.Timestamp `json:"accessExpirationTime"`
```
Comment (ymarcus93) [inline bastionzero/service/targets_disambiguated/all_targets.go:29]: Change this to `*types.Timestamp` because it can be null

**Review by ymarcus93 — APPROVED** [review]
LGTM. Will merge and release when prod is out

**Pull request merged**